src/main/util.py

```diff
@@ -17,7 +19,15 @@ def get_git_sha1(self, git_url: str, version: str) -> str:
     Returns:
         str: The SHA1 hash of the version.
     """
+    release_tag = requests.get(f"""https://api.github.com/repos/{git_url}/releases/tags/{version}""",
+                            timeout=10)
+    if release_tag.status_code != 200:
+        return ""
+    release_tag = release_tag.json()["tag_name"]
+    # Check if the response is successful
+    commit = requests.get(f"""https://api.github.com/repos/{git_url}/git/ref/tags/{release_tag}""",
+                          timeout=10)    
     sha1 = ""
-    # TODO
-    # Add function to get git sha1 number
+    if commit.status_code == 200:
+        sha1 = commit["sha"]
```

Comment:
This should be changed to ```sha1 =  commit["object"]["sha"]``` so that it matches the output stated by the [documentation](https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#get-a-reference).